libvast/src/view.cpp

```diff
@@ -164,4 +164,55 @@ data make_data(data_view x) {
   ), x);
 }
 
+// -- materialization ----------------------------------------------------------
+
+std::string materialize(std::string_view x) {
+  return std::string{x};
+}
+
+pattern materialize(pattern_view x) {
+  return pattern{std::string{x.string()}};
+}
+
+address materialize(address_view x) {
+  return address{x.data()};
+}
+
+subnet materialize(subnet_view x) {
+  return subnet{materialize(x.network()), x.length()};
+}
+
+namespace {
+
+std::pair<data, data> materialize(std::pair<data_view, data_view> x) {
+  return std::make_pair(materialize(x.first), materialize(x.second));
```

Comment:
I think `{materialize(x.first), materialize(x.second)}` would suffice as of C++14.